### DISCUSSION TITLE
Problema no curso de  Criando um App com React, TypeScript, Hooks, Context e Forms - Aula React Router

### DISCUSSION THREAD

**Discussion by victormoreiraa** [balta-io/forum#378] [Tira Dúvidas] [answered]
Boa tarde Professor, estava tudo ocorrendo bem até esta aula, aprece q ela pula etapas, antes dela não tinha nem feito o AddToDo.tsx.

tentei copiar do git mas deu varios erros.

ERROR in ./src/components/App.tsx 25:39-45
export 'Switch' (imported as 'Switch') was not found in 'react-router-dom' (possible exports: AbortedDeferredError, Await, BrowserRouter, Form, HashRouter, Link, MemoryRouter, NavLink, Navigate, NavigationType, Outlet, Route, Router, RouterProvider, Routes, ScrollRestoration, UNSAFE_DataRouterContext, UNSAFE_DataRouterStateContext, UNSAFE_LocationContext, UNSAFE_NavigationContext, UNSAFE_RouteContext, UNSAFE_useRouteId, UNSAFE_useScrollRestoration, createBrowserRouter, createHashRouter, createMemoryRouter, createPath, createRoutesFromChildren, createRoutesFromElements, createSearchParams, defer, generatePath, isRouteErrorResponse, json, matchPath, matchRoutes, parsePath, redirect, redirectDocument, renderMatches, resolvePath, unstable_HistoryRouter, unstable_useBlocker, unstable_usePrompt, useActionData, useAsyncError, useAsyncValue, useBeforeUnload, useFetcher, useFetchers, useFormAction, useHref, useInRouterContext, useLinkClickHandler, useLoaderData, useLocation, useMatch, useMatches, useNavigate, useNavigation, useNavigationType, useOutlet, useOutletContext, useParams, useResolvedPath, useRevalidator, useRouteError, useRouteLoaderData, useRoutes, useSearchParams, useSubmit)
ERROR in src/components/AddTodo.tsx:30:38
TS2558: Expected 0 type arguments, but got 1.
    28 |
    29 |     return (
  > 30 |         <form onSubmit={handleSubmit<AddTodoForm>(onSubmit)}>
       |                                      ^^^^^^^^^^^
    31 |             <h4>Nova tarefa</h4>
    32 |             <div className="uk-margin uk-width-1-1">
    33 |                 <input type="text" name="title" id="title" placeholder="Nova tarefa..." className="uk-input" ref={register} />
ERROR in src/components/AddTodo.tsx:33:110
TS2322: Type 'UseFormRegister<FieldValues>' is not assignable to type 'LegacyRef<HTMLInputElement> | undefined'.
  Type 'UseFormRegister<FieldValues>' is not assignable to type '(instance: HTMLInputElement | null) => void'.
    Types of parameters 'name' and 'instance' are incompatible.
      Type 'HTMLInputElement | null' is not assignable to type 'string'.
        Type 'null' is not assignable to type 'string'.
    31 |             <h4>Nova tarefa</h4>
    32 |             <div className="uk-margin uk-width-1-1">
  > 33 |                 <input type="text" name="title" id="title" placeholder="Nova tarefa..." className="uk-input" ref={register} />
       |                                                                                                              ^^^
    34 |                 
    35 |             </div>
    36 |             <div className="uk-width-1-1">
ERROR in src/components/App.tsx:7:10
TS2305: Module '"react-router-dom"' has no exported member 'Switch'.
     5 | import { BrowserRouter as Router,  Route } from 'react-router-dom';
     6 | import AddTodo from './AddTodo';
  >  7 | import { Switch } from 'react-router-dom';
       |          ^^^^^^
     8 |
     9 |
    10 | const App = () => {
    
    
    
alguns erros são dos pacotes q pediu para instalar

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @victormoreiraa, consegue colocar o trecho do seu código que este erro acontece?
